### PR TITLE
Shm18 kyrtaq

### PR DESCRIPTION
**Pull request by kyrtaq**

### CODE REVIEW

File: shm/fruit.hpp [hunk 1]
```diff
@@ -6,15 +6,18 @@ constexpr size_t expiryDate_{10};
...
     size_t getAmount() const override { return amount_; }
     size_t getBasePrice() const override { return basePrice_; }
     size_t getPrice() const override { return basePrice_ * (timeToExpire_ / expiryDate_); }
     size_t getTimeToExpire() const { return timeToExpire_; }
 
+    //override from Cargo
+    void nextDay() override;
+
 private:
+    Fruit& operator--();  // private? kyrtaq`s response: private!
```
Comment (NG90) [inline shm/fruit.hpp:20]: private should be ok for now. But where is the implementation of this operator?

File: shm/fruit.cpp [hunk 2]
```diff
@@ -23,3 +23,7 @@ bool Fruit::equals(const Cargo& rhs) const {
     return (name_ == rFruit->name_ &&
             timeToExpire_ == rFruit->timeToExpire_);
 }
+
+void Fruit::nextDay() {
+    Fruit::operator--();
```
Comment (NG90) [inline shm/fruit.cpp:28]: use `*this--` instead

File: shm/item.cpp [hunk 3]
```diff
@@ -12,3 +15,35 @@ bool Item::equals(const Cargo& rhs) const {
             basePrice_ == rItem->basePrice_ &&
             rarity_ == rItem->rarity_);
 }
+
+std::string Item::checkRarity() {
```
Comment (NG90) [inline shm/item.cpp:19]: maybe name `printRarity' would be more appropriate?

File: shm/ship.hpp [hunk 4]
```diff
@@ -25,12 +31,17 @@ class Ship {
...
     size_t maxCrew_{};
     size_t speed_{};
     const size_t id_{};
     std::string name_{};
     size_t capacity_{};
+    Delegate* delegatePlayer_{};
     size_t crew_{};
     std::vector<std::shared_ptr<Cargo>> cargo_{};
-};
+};
```
Comment (NG90) [inline shm/ship.hpp:47]: new line

**Comment by NG90** [comment]
TODO for @NG90: Think about changing delegate implementation, so `setDelegate()` is not needed.

File: shm/fruit.cpp (hunk 2 above)
Comment (kyrtaq) [reply shm/fruit.cpp:28]: @NG90 doesnt work

File: shm/fruit.hpp (hunk 1 above)
Comment (kyrtaq) [reply shm/fruit.hpp:20]: in the .cpp file

File: shm/item.cpp (hunk 3 above)
Comment (kyrtaq) [reply shm/item.cpp]: :+1:

File: shm/ship.hpp (hunk 4 above)
Comment (kyrtaq) [reply shm/ship.hpp]: :+1:

**Pull request closed without merging**